doc/charm++/manual.rst

```diff
@@ -5793,10 +5798,16 @@ futures, which include the following functions:
 | :code:`void CkSendToFuture(CkFuture fut, void *msg)` | :code:`void ck::future::set(T)`     |
 +------------------------------------------------------+-------------------------------------+
 
-You will note that the object-oriented versions are methods of `ck::future`,
-which can be templated with any pup'able type. An example of the
-object-oriented interface is available under `examples/charm++/future`,
-with an equivalent example for the C-compatible interface presented below:
+The object-oriented versions are methods of ``ck::future<T>``, which can be templated with any
+PUP-able type. Note, in most cases, messages/values cannot be retrieved via ``get`` when they were
+not been sent/set by a corresponding call to ``set``; however, it can receive messages of supported,
+internal message types sent via ``CkSendFuture``. Other message types must be wrapped as a PUP-able
+value and explicitly received as the expected message type(s); for example, one might wrap a message
```

Comment:
It would be nice to have an example or test program showing this

Reply:
Which part of this, more specifically? This is getting into "expert-user" territory and we document how to prepare internal messages elsewhere so, to some extent, it is probably OK to omit some documentation wrt using `CkSendFuture` as such. As for using `CkMarshalledMsg`, that's likely best documented elsewhere/in a separate PR, if it isn't documented already.